src/components/DonationWidget/index.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useEffect } from 'react';
+import { useMachine } from '@xstate/react';
+import donationWizardMachine from './machine';
+import {
+  DonationMachineContext,
+  DonationMachineStateValueMap
+} from './machine/types';
+// TODO: avoid to use AnyEventObject in favor of DonationMachineEvent
+import { AnyEventObject } from 'xstate';
+
+export interface DonationWidgetProps {
+  /**
+   * Optional identifier for the widget
+   */
+  id?: string;
+}
+
+const DonationWidget: React.FC<DonationWidgetProps> = ({ id }) => {
+  const [state, send] = useMachine<DonationMachineContext, AnyEventObject>(
+    donationWizardMachine
+  );
+  const { context: machineCtx } = state;
+  const machineState: DonationMachineStateValueMap = state.value;
+
+  useEffect(() => {
+    if (machineState === 'failure') {
+      alert(`Something went wrong ${JSON.stringify(machineCtx)}`);
+      send('RETRY');
+    }
+  });
+
+  const handleSubmitAmountForm = (e: React.ChangeEvent<HTMLFormElement>) => {
+    const data = new FormData(e.currentTarget);
+    const value = Number(data.get('amount'));
+
+    if (!value) return;
+
+    send([
+      {
+        type: 'UPDATE.AMOUNT',
+        value
+      },
+      { type: 'NEXT' }
+    ]);
+    e.preventDefault();
+  };
+
+  const handleSubmitPaymentInfoForm = (
+    e: React.ChangeEvent<HTMLFormElement>
+  ) => {
+    const data = new FormData(e.currentTarget);
+    const paymentInformation = {
+      firstName: data.get('first-name'),
+      lastName: data.get('last-name'),
+      email: data.get('email'),
+      cardNumber: data.get('card')
+    };
+
+    send({
+      type: 'NEXT',
+      ...paymentInformation
+    });
+    e.preventDefault();
+  };
+
+  const handleSubmitAddressForm = (e: React.ChangeEvent<HTMLFormElement>) => {
+    const data = new FormData(e.currentTarget);
+    const addressInformation = {
+      address: data.get('address'),
+      city: data.get('city'),
+      zipCode: data.get('zipCode'),
+      country: data.get('country')
+    };
+
+    send({
+      type: 'NEXT',
+      ...addressInformation
+    });
+    e.preventDefault();
+  };
+
+  console.log(state);
+  return (
+    <div id={id} className="m-auto" style={{ width: '420px' }}>
```

Comment:
why did you use a fixed width instead of a tailwind style?

Reply:
For sure we will use tailwind, at this point I was only laying up the idea, that's why this file have all the forms in sort of raw way to do things.